.github/workflows/build.yml

```diff
@@ -0,0 +1,61 @@
+name: Build Presentations
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+          
+      - name: Get pnpm store directory
```

Comment:
This is an old way. Read and follow https://github.com/pnpm/action-setup?tab=readme-ov-file#use-cache-to-reduce-installation-time